### PR TITLE
Fixed wrong unit initialization: will now work with Celsius/Fahrenheit

### PR DESCRIPTION
**Pull request by JochenHiller**
Signed-off-by: Jochen Hiller <j.hiller@telekom.de>

### CODE REVIEW

File: binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java [hunk 1]
```diff
@@ -64,8 +64,13 @@ public void initialize() {
...
 
         location = (String) config.get("location");
-        if ("us".equalsIgnoreCase((String) config.get("unit"))) {
+        if ("metric".equalsIgnoreCase((String) config.get("unit"))) {
+            unit = "c";
+        } else if ("us".equalsIgnoreCase((String) config.get("unit"))) {
             unit = "f";
+        } else {
+            // means unit not known
+            unit = null;
```
Comment (kaikreuzer) [inline binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java]: I am not sure if this is the right thing to do. Imho we should either
a) cancel the initialization as there is an invalid configuration
b) ignore an invalid unit parameter and go for the default, which is "metric" as per the thing type definition. But the implementation now uses the default of Yahoo weather, which is fahrenheit.

Comment (dnobel) [inline binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java]: a) the Binding could throw an exception in that case, but then the thing it is not usable. Maybe we can think of the correct behavior when we introduce the additional status and extend the binding again.
b) as we agreed to write the default value into the configuration, there is no need to apply the default again. If the user does not override the value explicitly to null, the configuration will automatically include the default value. So there is no need of default handling inside handlers any more.

Comment (kaikreuzer) [inline binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java]: a) yes, once we have the new "MIS/NOT-CONFIGURED" status, we should use it here.
b) wrong. This line in the code deals with the fact that the config property is invalid for some reason. So the default is NOT applied by the framework and the code deliberately sets the unit to "null", which results in requests that return fahrenheit values.

Comment (dnobel) [inline binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java]: OK, you are right. But in general it means, that it should set the thing into the status "MISCONFIGURED". So we can change the binding, after we introduced the new status.

**Pull request merged**